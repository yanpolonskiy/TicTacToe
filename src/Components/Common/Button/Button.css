.button {
  height: 36px;
  padding: 8px 16px;
  background-color: white;
  border: 1px solid darkslategrey;
  border-radius: 18px;
}
