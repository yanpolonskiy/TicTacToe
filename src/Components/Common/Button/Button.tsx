import React from "react";
import "./Button.css";

/**
 *
 * @param props
 * @constructor
 */
export const Button = (props: React.HTMLProps<HTMLButtonElement>) => {
  const className = "button " + (props.className ? props.className : "");
  return <button {...props} type="button" className={className} />;
};
