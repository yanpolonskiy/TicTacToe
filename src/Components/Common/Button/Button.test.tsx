import React from 'react';
import {Button} from './Button';
import renderer from 'react-test-renderer';

/**
 * Тест отрисовки кнопки.
 */
test("renders button", () => {
    const tree = renderer
    .create(<Button>Тестовая кнопка</Button>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});
