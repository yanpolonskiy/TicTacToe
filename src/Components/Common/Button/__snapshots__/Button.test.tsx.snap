// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button 1`] = `
<button
  className="button "
  type="button"
>
  Тестовая кнопка
</button>
`;
